pkg/config/config.go

```diff
@@ -125,3 +131,22 @@ func mkdir(path string) error {
 func getFullKey(env, path string) string {
 	return KeyEnvironment + "." + env + "." + path
 }
+
+// validateKey validates the key against the following rules:
+// 1. key must start with a word character
+// 2. key must contain only word characters, dashes or dots
+// 3. key must end with a word character
+func validateKey(key string) error {
+	pattern := `^\w[\w\-\.]*\w$`
```

Comment:
Pattern should probably not start with a number but I'd do some further validation that the string is actually something we can use and that the value works for the key.

Reply:
udpated the regex so it now expects pattern to start and end with letters

**Value validation**: since tclt-kit's config feature is mostly generic, seems we can validate only the values for the known keys such as `env`, `current-env`?

If we want to validate value types for the other tctl options, this would be done in tctl itself